### PR TITLE
web v1

### PR DESCRIPTION
**Pull request by jenikm**
https://github.com/enova/saigo/blob/master/exercise-003-web/README.md
Adding persistent web server with tests

### CODE REVIEW

File: exercise-003-web/utils.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+func SaveState(users *map[string]int, dbPath string) {
```
Comment (irshadaj) [inline exercise-003-web/utils.go]: I might be missing something here- is there a reason why Mount(), Unmount(), SaveState(), RestoreState() and Exists() are all exported functions?

File: exercise-003-web/handlers.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	}
+}
+
+func Mount() {
+	setup("./templates")
+	RestoreState(&users, DbPath)
+}
+
+func Unmount() {
+	SaveState(&users, DbPath)
```
Comment (irshadaj) [inline exercise-003-web/handlers.go]: I'd get rid of Unmount and just call SaveState() wherever needed.

File: exercise-003-web/handlers.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+func HandleUsers(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodPost:
+		handlePostUsers(w, r)
+	default:
+		http.Error(w, "", http.StatusNotFound)
+	}
+}
+
+func Mount() {
```
Comment (irshadaj) [inline exercise-003-web/handlers.go]: If this is only being used in one place, I'd skip making this a function by itself.

File: exercise-003-web/utils.go (hunk 1 above)
Comment (jenikm) [reply exercise-003-web/utils.go]: @irshadaj Thank you for your feedback.

I organized code into three sections:
1. Server - starts/stops server
2. Handlers - logic for users, templates rendering, presenters, stores application state
3. Utils - file utils, logic how to serialize / de-serialize given object into JSON.

Ideally, I would like to keep all those pieces separate somehow. Please let me know if you have pattern of separating responsibility and also conform to Golang standards.

Comment (irshadaj) [reply exercise-003-web/utils.go]: In cases where you don't want/need multiple packages, you could put the pieces in separate files. Doing this makes no difference to the compiler- all the code in a single package is treated as if it were in one place; but you'll still have the logical separation that you desire. You can also make all the functions unexported.

**Pull request closed without merging**